### PR TITLE
Fix typos prior to release

### PR DESCRIPTION
**Pull request by tom-cosgrove-arm**
## Gatekeeper checklist

- [x] **changelog** not required - minor typos
- [x] **backport** #6718
- [x] **tests**  not required (I wish there was a good way of automating this)

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
This conflicts with #6689. Please don't touch changelog entries here or rebase on top of #6689. Likewise with backports.

**Comment by tom-cosgrove-arm** [comment]
> Likewise with backports

Not sure what "likewise" means here. The backport #6718 doesn't touch any changelog entries. Or does "likewise" here mean "don't backport"? I thought it would make more sense to have typos fixed in both branches so that future backport PRs apply cleanly.

Can you clarify?

**Review by davidhorstmann-arm — DISMISSED** [review]
LGTM

File: library/ssl_tls12_client.c [hunk 1]
```diff
@@ -1972,7 +1972,7 @@ static int ssl_parse_server_psk_hint( mbedtls_ssl_context *ssl,
 
     /*
      * Note: we currently ignore the PKS identity hint, as we only allow one
```
Comment (wernerlewis) [inline library/ssl_tls12_client.c:1974]:
```suggestion
     * Note: we currently ignore the PSK identity hint, as we only allow one
```

Comment (tom-cosgrove-arm) [reply library/ssl_tls12_client.c]: Nice spot - thanks - updated here and in the backport (sorry @davidhorstmann-arm )

**Review by davidhorstmann-arm — APPROVED** [review]

**Comment by tom-cosgrove-arm** [comment]
CI is green

**Review by wernerlewis — APPROVED** [review]

**Pull request merged**